### ISSUE TITLE
radiff2 has a weird behaviour

### ISSUE THREAD

**Issue by daniel-rome** [radare/radare2#5142]
The radiff2 tool always returns a 1.0 "Percentage of matching” value, even if the function has been modified:

PoC with two similar binaries:
```
$ cat a.c
#include <stdio.h>
int main(int argc, char *argv[]) {
 int value = 10;
 printf("%d\n", value);
}
---
$ cat b.c 
#include <stdio.h>
int main(int argc, char *argv[]) {
 int value = 20;
 printf("%d\n", value);
}
```
Once the binaries have been compiled and a diffing is done using the radiff2 tool, the results can be seen below:
```
$ radiff2 -A -C a.out b.out 
<snip>
                   entry0  42 0x400440 |   MATCH  (1.000000) | 0x400440   42 entry0
sym.imp.__libc_start_main  48 0x400420 |   MATCH  (1.000000) | 0x400420   48 sym.imp.__libc_start_main
                 sym.main  44 0x40052d | UNMATCH  (1.000000) | 0x40052d   44 sym.main
           sym.imp.printf  32 0x400410 |   MATCH  (1.000000) | 0x400410   32 sym.imp.printf
 sym.deregister_tm_clones  41 0x400470 |   MATCH  (1.000000) | 0x400470   41 sym.deregister_tm_clones
   sym.register_tm_clones  57 0x4004a0 |   MATCH  (1.000000) | 0x4004a0   57 sym.register_tm_clones
sym.__do_global_dtors_aux  28 0x4004e0 |   MATCH  (1.000000) | 0x4004e0   28 sym.__do_global_dtors_aux
          sym.frame_dummy  42 0x400500 |   MATCH  (1.000000) | 0x400500   42 sym.frame_dummy
      sym.__libc_csu_fini   2 0x4005d0 |   MATCH  (1.000000) | 0x4005d0    2 sym.__libc_csu_fini
                sym._fini   9 0x4005d4 |   MATCH  (1.000000) | 0x4005d4    9 sym._fini
      sym.__libc_csu_init 101 0x400560 |   MATCH  (1.000000) | 0x400560  101 sym.__libc_csu_init
                sym._init  26 0x4003e0 |   MATCH  (1.000000) | 0x4003e0   26 sym._init
   loc.imp.__gmon_start__  48 0x400430 |   MATCH  (1.000000) | 0x400430   48 loc.imp.__gmon_start__
```
As the above code shows, the "main" function has changed, but the "Percentage of matching” is the same.

**Comment by radare** [comment]
Because radiff skips constant values because diffing with relocated code shouldnt affect the result. The thing is.. How do you differentiate an immediate with a relocated value?

**Comment by daniel-rome** [comment]
Anyways, using the following examples, we get the same result:
```
$ cat a.c 
#include <stdio.h>
int main(int argc, char *argv[]) {
 char a[10] = "b\0";
 char b[10] = "\0";
 printf("%s\n", b);
 strcpy(b, a);
 printf("%s\n", b);
}
---
$ cat b.c
#include <stdio.h>
int main(int argc, char *argv[]) {
 char a[10] = "b\0";
 char b[10] = "\0";
 int c = 5;
 printf("%s\n", b);
 strncpy(b, a, c);
 printf("%s\n", b);
}
```

Result:
```
radiff2 -A -C ./a.out ./b.out
<split>
                   entry0  42 0x400500 | UNMATCH  (1.000000) | 0x400500   42 entry0
sym.imp.__libc_start_main  48 0x4004e0 |   MATCH  (1.000000) | 0x4004e0   48 sym.imp.__libc_start_main
                 sym.main 152 0x4005ed | UNMATCH  (1.000000) | 0x4005ed  171 sym.main
             sym.imp.puts  48 0x4004c0 |   MATCH  (1.000000) | 0x4004c0   48 sym.imp.puts
<split>
```

**Comment by radare** [comment]
yeah that doesnt seems good

@jroimartin wanna take a look?

**Comment by Maijin** [comment]
May be related with #5362

**Comment by computereasy** [comment]
@Maijin  @daniel-rome  Hello, I am still confused with the representation. How does 1.00000 mean regarding "**unmatch**" ??

**Comment by radare** [comment]
yes i think this is a regression after merging the new diffing algorithm changes. we can hopefully fix this inconsistency in the next iteration. moving to 0.10.6

**Comment by radare** [comment]
I fixed this last week. fill another issue if you found any other inconsistency
